src/main/java/org/cyclops/integrateddynamics/core/evaluate/operator/CombinedOperator.java

```diff
@@ -253,6 +254,54 @@ public CombinedOperator newFunction(IOperator... operators) throws EvaluationExc
         }
     }
 
+    public static class Duplicate extends OperatorsFunction {
+
+        public Duplicate(IOperator operator) {
+            super(new IOperator[]{operator});
+        }
+
+        @Override
+        public IValue evaluate(SafeVariablesGetter variables) throws EvaluationException {
+            IOperator operator = getOperators()[0];
+            if(operator.getRequiredInputLength() == 1) { // TODO ValueHelpers.evaluateOperator should do this uncurrying, instead of having to do it here. issue #352
+                IValue value = variables.getValue(0);
+                IValue operatorValue = ValueHelpers.evaluateOperator(operator, value);
+                if(!ValueHelpers.correspondsTo(operatorValue.getType(), ValueTypes.OPERATOR)) {
+                    throw new EvaluationException(String.format("Intermediate value was of type %s, expected Operator", operatorValue.getType().getTypeName()));
+                }
+                operator = ((ValueOperator)operatorValue).getRawValue();
+                return ValueHelpers.evaluateOperator(operator, value);
+            } else {
+                IValue[] values = { variables.getValue(0), variables.getValue(0) };
+                return ValueHelpers.evaluateOperator(operator, values);
+            }
+        }
+
+        public static CombinedOperator asOperator(IOperator operator) throws EvaluationException {
+            CombinedOperator.Duplicate duplicate = new CombinedOperator.Duplicate(operator);
+            IValueType[] duplicatedInputTypes = new IValueType[] {operator.getInputTypes()[0]};
+            try {
+                return new CombinedOperator("W", "duplicated", duplicate, duplicatedInputTypes,
+                        operator.getOutputType(), IConfigRenderPattern.PREFIX_1);
```

Comment:
I'm worried that `operator.getOutputType()` may be incorrect in the case that uncurrying happens.